### ISSUE TITLE
Divides a b :- (Mod b a ~ 0)

### ISSUE THREAD

**Issue by Bodigrim** [ekmett/constraints#96]
I was trying to derive `Divides a b :- (Mod b a ~ 0)` and - against my expectations - it appeared to be quite convoluted. 

```haskell
foo :: forall a b. (Divides a b, 1 <= a) :- (Mod b a ~ 0)
foo = Sub $ case (dividesDef @a @b, euclideanNat @a @b, plusIsCancellative @b @0 @(Mod b a)) of
  (Sub Dict, Sub Dict, Sub Dict) -> Dict
```

This is quite frustrating, because initially I expected `Mod b a ~ 0` to be the very definition of `Divides`. Is there a good reason to define it as `type Divides n m = n ~ Gcd n m`? (This also means `Divides 0 0`, which is wrong).

I see three ways to improve UX here:

1. (cleanest) Define `type Divides a b = (Mod b a ~ 0, 1 <= a)` and add an axiom `Divides n m :- n ~ Gcd n m`. This arguably defines `Divides` in the most natural way and fully in terms of vanilla GHC type families (no `magic` involved), and also fixes `Divides 0 0`. 

2. Leave `type Divides n m = n ~ Gcd n m` in peace, but change 

```diff
- dividesDef :: forall a b. Divides a b :- ((a * Div b a) ~ b)
+ dividesDef :: forall a b. Divides a b :- (Mod b a ~ 0)
```

Here the previous behaviour of `dividesDef` can be easily restored by substituting it into `euclideanNat`. The cons are that this is still a breaking change, and makes `Divides 0 0` even worse, because it now leads to `Mod 0 0 ~ 0`.

3. (least intrusive) Just add `(Divides a b, 1 <= a) :- (Mod b a ~ 0)` to `Data.Constraint.Nat`, so that future users would not need to reinvent it themselves. 

My personal preference is option 1.

**Comment by Bodigrim** [comment]
Ping @RyanGlScott @ekmett

**Comment by ekmett** [comment]
Option 2 in your list is pretty clean.

**Comment by RyanGlScott** [comment]
Fixed in #98.
